### PR TITLE
Add operator for combining two alternatives.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
Moved from https://github.com/purescript-contrib/purescript-parsing/pull/43

### CODE REVIEW

File: src/Data/Either.purs [hunk 1]
```diff
@@ -224,6 +224,10 @@ either :: forall a b c. (a -> c) -> (b -> c) -> Either a b -> c
 either f _ (Left a) = f a
 either _ g (Right b) = g b
 
+-- | Combine two alternatives.
+eitherP :: forall m a b. Alt m => m a -> m b -> m (Either a b)
+eitherP a b = (Left <$> a) <|> (Right <$> b)
```
Comment (paf31) [inline src/Data/Either.purs]: Nitpick: the parens are redundant here.

**Comment by paf31** [comment]
Thanks!

File: src/Data/Either.purs (hunk 1 above)
Comment (tmcgilchrist) [inline src/Data/Either.purs]: Removed.

**Comment by paf31** [comment]
I renamed this to `choose` actually, since `eitherP` seemed a bit cryptic, since this is no longer specific to parsers.

**Pull request merged**